### PR TITLE
feat: implement file mode

### PR DESCRIPTION
**Pull request by naman-bruno**
#### Contribution Checklist:

- [ ] **I've used AI significantly to create this pull request**
- [ ] **The pull request only addresses one issue or adds one feature.**
- [ ] **The pull request does not introduce any breaking changes**
- [ ] **I have added screenshots or gifs to help explain the change if applicable.**
- [ ] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **New Features**
  * Added a collection file mode toggle to switch between file editing and request management.
  * Introduced a code editor experience for viewing and editing raw file content.
  * Implemented save support for file-mode drafts with success/error notifications via a dedicated error modal.
* **Enhancements**
  * Enabled autosave behavior for file-mode drafts to persist edits automatically.
* **Tests**
  * Expanded coverage for file-mode reducers and draft/raw synchronization edge cases.

### CODE REVIEW

**Review by gopu-bruno — APPROVED** [review]

File: packages/bruno-app/src/components/FileEditor/index.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        onEdit={onEdit}
+        onSave={onSave}
+        toggleFileMode={_toggleFileMode}
+        mode={editorMode}
+        font={get(preferences, 'font.codeFont', 'default')}
+        initialScroll={scroll}
+        onScroll={setScroll}
+      />
+      <IconDeviceFloppy
+        onClick={hasChanges ? onSave : undefined}
```
Comment (bijin-bruno) [inline packages/bruno-app/src/components/FileEditor/index.js]: I think this should be handled within the onSave handler.

```
  const onSave = () => {
    if (!hasChanges) return;

    dispatch(saveFile(content, item?.uid, collection?.uid));
  };

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js [hunk 2]
```diff
@@ -193,6 +194,59 @@ export const saveRequest = (itemUid, collectionUid, silent = false) => (dispatch
   });
 };
 
+export const saveFile = (content, itemUid, collectionUid, silent = false) => async (dispatch, getState) => {
```
Comment (bijin-bruno) [inline packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js:197]: in EE saveFile returns a promise. Let's keep the code in sync? Or is there a specific reason for the change?

**Review by bijin-bruno — CHANGES_REQUESTED** [review]

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js (hunk 2 above)
Comment (naman-bruno) [reply packages/bruno-app/src/providers/ReduxStore/slices/collections/actions.js:197]: this is done because of this comment: https://github.com/usebruno/bruno/pull/8258#discussion_r3405003285

**Review by bijin-bruno — APPROVED** [review]
